src/main/java/com/example/internalbookingmanagement/entity/BaseUser.java

```diff
@@ -0,0 +1,60 @@
+package com.example.internalbookingmanagement.entity;
+
+import com.example.internalbookingmanagement.entity.enums.Role;
+import jakarta.persistence.*;
+import lombok.*;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.List;
+
+@Entity
+@Getter
+@Setter
+@AllArgsConstructor
+@Builder
+@NoArgsConstructor
+
+public class BaseUser extends AbstractEntity implements UserDetails {
+
+    private String status;
+    private String userName;
+    private String password;
+    private String email;
+    @ElementCollection(fetch = FetchType.EAGER)
+    @CollectionTable(name = "user_roles", joinColumns = @JoinColumn(name = "user_id"))
+    @Column(name = "role")
+    @Enumerated(EnumType.STRING)
+    private List<Role> roles;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return roles;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return true;
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return true;
+    }
+
+    @Override
+    public String getUsername() {
+        return null;
```

Comment:
Please, return a valid username here. Replace return null with return this.username;